frontend/src/utils/store-sync.ts

```diff
@@ -1,161 +1,206 @@
 import { DisplayPreferencesDto } from '@jellyfin/sdk/lib/generated-client';
 import { getDisplayPreferencesApi } from '@jellyfin/sdk/lib/utils/api/display-preferences-api';
 import destr from 'destr';
-import { isNil } from 'lodash-es';
 import { usei18n, useRemote, useSnackbar } from '@/composables';
 import { taskManagerStore } from '@/store';
 
+const CLIENT = 'vue';
+
+const USER_UNAUTHENTICATED_MESSAGE = 'Expected user to be authenticated';
+
 /**
- * Cast custom preferences returned from the server from strings to the correct Javascript type
- *
- * @param data - Response from the server
- * @param store
+ * Serializes custom pref values for storage as string
  */
-function castDisplayPreferencesResponse<T>(
-  data: DisplayPreferencesDto,
-  target: T
-): DisplayPreferencesDto {
-  // @ts-expect-error - We need loose types here since all this logic is about properly casting the data
-  const storeKeys = Object.keys(target);
-
-  if (data.CustomPrefs) {
-    for (const [key, value] of Object.entries(data.CustomPrefs)) {
-      /**
-       * destr does proper conversion for all the types, even undefined
-       */
-      if (storeKeys.includes(key)) {
-        data.CustomPrefs[key] = destr(value);
-      } else {
-        /**
-         * Remove properties that are automatically present in the server but are not present in the store.
-         */
-        delete data.CustomPrefs[key];
-      }
-    }
+function serializeCustomPref(value: unknown): string | undefined {
+  if (value === undefined) {
+    return undefined;
   }
 
-  return data.CustomPrefs || {};
+  return typeof value === 'string' ? value : JSON.stringify(value);
+}
+
+/**
+ * De-serializes custom pref values from string to a value
+ */
+function deserializeCustomPref(value: string): unknown {
+  return destr(value);
 }
 
 /**
- * Fetch displayPreferences settings from server
- *
- * @param key - Key used to save the store's state
- * @param target - Object that serves as a model for the store's state
+ * Fetches server display preferences
  */
-export async function fetchSettingsFromServer<T>(
-  key: string,
-  target: T,
-  cast = true
-): Promise<T | DisplayPreferencesDto> {
+export async function fetchDisplayPreferences(
+  displayPreferencesId: string
+): Promise<DisplayPreferencesDto> {
   const remote = useRemote();
 
+  if (!remote.auth.currentUserId) {
+    throw new Error(USER_UNAUTHENTICATED_MESSAGE);
+  }
+
```

Comment:
This check isn't really done anywhere else, we usually just default the currentUserId if it's null, so I'll likely just do that instead of this error check everywhere.